### PR TITLE
Sdc cxx conversion for non-f_rhs sdc_util.F90 functions

### PR DESCRIPTION
**Pull request by jmsexton03**
## PR summary

Update spacing to match conventions
Make transition sdc_util.F90 and sdc_vode.F90 counterparts
Update convergence bash scripts to match current flags
Convert non-f_rhs and burn_state depending sdc fortran functions

## PR checklist

- [X] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [ ] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by jmsexton03** [comment]
The conflict looks like it's due to removing a file:
`CONFLICT (modify/delete): Exec/hydro_tests/acoustic_pulse/convergence_rk4.sh deleted in development and modified in HEAD. Version HEAD of Exec/hydro_tests/acoustic_pulse/convergence_rk4.sh left in tree.`

**Comment by zingale** [comment]
retested and all tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-04-06-006/index.html

**Comment by zingale** [comment]
ran the `reacting_convergence` test on this PR.  Here are the L1 norms for SDC-4:

64-128-256:
```
 Level  L1 norm of Error in Each Component
-----------------------------------------------
level: 0
Warning: BoxArray lengths are not the same at level 0
                 density        2.127414e+17       3.855180   1.470034e+16
                    xmom        3.401348e+25       3.855819   2.349276e+24
                    ymom        3.401348e+25       3.855819   2.349276e+24
                    zmom        0.000000e+00   ------------   0.000000e+00
                   rho_E        1.945135e+34       3.891495   1.310669e+33
                   rho_e        1.671836e+34       3.899408   1.120353e+33
                    Temp        1.236297e+20       3.707660   9.462506e+18
                 rho_He4        2.147386e+17       3.857502   1.481447e+16
                 rho_C12        8.788506e+15       3.797869   6.318918e+14
                 rho_O16        1.294143e+13       3.765158   9.518247e+11
                rho_Fe56        2.127543e-13       3.853675   1.471657e-14
```

128-256-512:
```
Level  L1 norm of Error in Each Component
-----------------------------------------------
level: 0
                 density        1.470034e+16       3.971882   9.368540e+14
                    xmom        2.349276e+24       3.958186   1.511476e+23
                    ymom        2.349276e+24       3.958186   1.511476e+23
                    zmom        0.000000e+00   ------------   0.000000e+00
                   rho_E        1.310669e+33       3.952910   8.463470e+31
                   rho_e        1.120353e+33       3.955271   7.222704e+31
                    Temp        9.462506e+18       3.949348   6.125395e+17
                 rho_He4        1.481447e+16       3.969372   9.457710e+14
                 rho_C12        6.318918e+14       3.910981   4.200686e+13
                 rho_O16        9.518247e+11       3.871895   6.501303e+10
                rho_Fe56        1.471657e-14       3.966652   9.412939e-16
```

File: Exec/science/flame/convergence_strang.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 set -x
 
 DIM=1
-EXEC=./Castro${DIM}d.gnu.MPI.ex
+EXEC=./Castro${DIM}d.gnu.MPI.TRUESDC.ex
```
Comment (zingale) [inline Exec/science/flame/convergence_strang.sh:7]: this is meant to be Strang+CTU, so there is no suffix `TRUESDC` needed

**Comment by zingale** [comment]
`acoustic_pilse_general` convergence test (SDC-4)

64-128-256:
```
Level  L1 norm of Error in Each Component
-----------------------------------------------
level: 0
                 density        1.935153e+17       3.939203   1.261529e+16
                    xmom        3.842240e+25       3.906855   2.561557e+24
                    ymom        3.842240e+25       3.906855   2.561557e+24
                    zmom        0.000000e+00   ------------   0.000000e+00
                   rho_E        4.078578e+34       3.939004   2.659196e+33
                   rho_e        3.525830e+34       3.949177   2.282658e+33
                    Temp        5.656932e+19       3.954672   3.648430e+18
                   rho_X        1.935153e+17       3.939203   1.261529e+16
```

128-256-512:
```
Level  L1 norm of Error in Each Component
-----------------------------------------------
level: 0
                 density        1.261529e+16       3.981280   7.987530e+14
                    xmom        2.561557e+24       3.971640   1.632756e+23
                    ymom        2.561557e+24       3.971640   1.632756e+23
                    zmom        0.000000e+00   ------------   0.000000e+00
                   rho_E        2.659196e+33       3.981310   1.683668e+32
                   rho_e        2.282658e+33       3.982289   1.444283e+32
                    Temp        3.648430e+18       3.990836   2.294798e+17
                   rho_X        1.261529e+16       3.981280   7.987530e+14
```

File: Source/sdc/Castro_sdc.cpp [hunk 2]
```diff
@@ -5,350 +5,375 @@
...
+                                    AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+
+            // now do the conservative update using this <R> to get <U>
+            // We'll also need to pass in <C>
+            ca_sdc_conservative_update(bx, dt_m, k_new_m_start_arr, k_new_m_end_arr,
+                                       C_source_arr,R_new_arr);
+
+        }
+#else
+        amrex::Array4<const amrex::Real> const& k_new_m_start_arr=
```
Comment (zingale) [inline Source/sdc/Castro_sdc.cpp:250]: you might consider doing `using namespace amrex` and dropping all of the `amrex::`

File: Source/sdc/sdc_util.cpp [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+    // Gauss-Lobatto (Simpsons)
+
+    if (m_start == 0)
+    {
+        AMREX_PARALLEL_FOR_4D(bx, k_n.nComp(), i, j, k, n,
+        {
+            k_n(i,j,k,n) = k_m(i,j,k,n) +
+                dt_m * (A_m(i,j,k,n) - A_0_old(i,j,k,n)) +
+                dt/1800.0_rt * ((-35.0_rt*sqrt(6.0_rt) + 440.0_rt)*A_1_old(i,j,k,n) +
```
Comment (zingale) [inline Source/sdc/sdc_util.cpp:137]: should use `std::sqrt` to be safe

File: Source/sdc/sdc_util.cpp [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+
+    if (m_start == 0)
+    {
+        // compute the integral from [t_m, t_{m+1}], normalized by dt_m
+        AMREX_PARALLEL_FOR_4D(bx, C.nComp(), i, j, k, n,
+        {
+            Real integral = (dt/dt_m) * (1.0_rt/1800.0_rt) *
+                ((-35.0_rt*sqrt(6.0_rt) + 440.0_rt)*(A_1_old(i,j,k,n) + R_1_old(i,j,k,n)) +
+                 (-169.0_rt*sqrt(6.0_rt) + 296.0_rt)*(A_2_old(i,j,k,n) + R_2_old(i,j,k,n)) +
+                 (-16.0_rt + 24.0_rt*sqrt(6.0_rt))*(A_3_old(i,j,k,n) + R_3_old(i,j,k,n)));
```
Comment (zingale) [inline Source/sdc/sdc_util.cpp:321]: `std::sqrt`

File: Source/sdc/sdc_util.cpp [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+void Castro::ca_store_reaction_state(const amrex::Box& bx,
+                                     amrex::Array4<const amrex::Real> const& R_old,
+                                     amrex::Array4<const amrex::Real> const& state,
+                                     amrex::Array4<amrex::Real> const& R_store)
+{
+    // copy the data from the last node's reactive source to the state data
+
+    // for R_store we use the indices defined in Castro_setup.cpp for
+    // Reactions_Type
+    int nspec = R_store.nComp()-2;
```
Comment (zingale) [inline Source/sdc/sdc_util.cpp:415]: it is safer to use `NumSpec` from the `network.H`

Comment (jmsexton03) [reply Source/sdc/sdc_util.cpp:415]: It looks like that's a `constexpr int`, which should be usable inside the lambda. Is it cleaner to just use `NumSpec` directly, or should I keep using the locally scoped copy?

Comment (zingale) [reply Source/sdc/sdc_util.cpp:415]: I think you should just use NumSpec directly
that's what we do elsewhere in the code.

**Pull request merged**